profiles/common/modules/features/nexteuropa_newsroom/classes/NewsroomAgenda.inc

```diff
@@ -0,0 +1,679 @@
+    $this->query = $value;
+  }
+
+  /**
+   * Getter.
+   */
+  public function getResult() {
+    return $this->result;
+  }
+
+  /**
+   * Setter.
+   */
+  public function setResult($value) {
+    $this->result = $value;
+  }
+
+  /**
+   * Setter.
+   */
+  public function setTypeId($value) {
+    $this->typeId = $value;
+  }
+
+  /**
+   * Getter.
+   */
+  public function getTypeId() {
+    return $this->typeId;
+  }
+
+  /**
+   * Setter.
+   */
+  public function setTopics($value) {
+    $this->topics = $value;
+  }
+
+  /**
+   * Getter.
+   */
+  public function getTopics() {
+    return $this->topics;
+  }
+
+  /**
+   * Add item count.
+   */
+  public function addItemsCount() {
+    $this->itemsCount++;
+  }
+
+  /**
+   * Clear URL parts.
+   */
+  public function clearUrlParts() {
+    $this->urlParts = array();
+  }
+
+  /**
+   * Adds URL part.
+   *
+   * @param string $value
+   *   URL part.
+   */
+  public function addUrlPart($value) {
+    $this->urlParts[] = $value;
+  }
+
+  /**
+   * NewsroomAgenda constructor.
+   *
+   * @param string $type
+   *   Newsroom type.
+   * @param array $topics
+   *   Newsroom topics.
+   */
+  public function __construct($type, $topics = array()) {
+    if ($type) {
+      $this->setTypeId($type->tid);
+      $this->setTypeUrl($type->name);
+    }
+
+    if (is_object($topics)) {
+      $this->setTopicUrl($topics->name);
+      $topics = array($topics);
+    }
+
+    $this->setTopics($topics);
+
+    $this->addUrlPart($this->getView());
+    $this->addUrlPart($this->getTypeUrl());
+    $this->addUrlPart($this->getTopicUrl());
+  }
+
+  /**
+   * Prepare agenda components: query, result, the dates range.
+   */
+  protected function prepareAgenda() {
+    $this->getBaseQuery();
+    $this->setResult($this->prepareResult($this->loadResult()));
+    $this->setupAgenda();
+  }
+
+  /**
+   * Returns previous title.
+   *
+   * @return string
+   *   Title
+   */
+  public function getPreviousTitle() {
+    return t('Previous');
+  }
+
+  /**
+   * Returns next title.
+   *
+   * @return string
+   *   Title
+   */
+  public function getNextTitle() {
+    return t('Next');
+  }
+
+  /**
+   * Returns form id with the current ID.
+   *
+   * @param string $form_id
+   *   Form ID.
+   *
+   * @return array
+   *   Form array.
+   */
+  protected function getForm($form_id) {
+    $topics = $this->getTopics();
+    $topic_id = count($topics) == 1 ? $topics[0]->tid : NULL;
+    $result_topics = $form_id == 'nexteuropa_newsroom_agenda_filter_form' ? $this->getResultTopicsId() : NULL;
+    return drupal_get_form($form_id, $this->getView(), $this->getTypeId(), $topic_id, $this->getCurrentDate()->format('Y-m-d'), $result_topics);
+  }
+
+  /**
+   * Returns the agenda html.
+   *
+   * @return string
+   *   HTML.
+   */
+  public function getAgenda() {
+    $cache_key = 'get_agenda:page:' . $this->getUniqueCachePart();
+    $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
```

Comment:
It cannot be cached because the template has form rendering what always will clear it. And actually the idea to put a full tpl to cache is also quite weird because it has a lot of dynamic content what should be rendered/generated on the fly.